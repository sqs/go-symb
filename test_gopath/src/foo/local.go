package foo

var NonLocalVar = 1

type NonLocalType int

func (localRecv *NonLocalType) NonLocalFunc(localParam int) (localResult int) {
	var localVar int = 1
	println(NonLocalVar, localRecv.NonLocalFunc, NonLocalType(3), localVar, localParam, localResult)
	return localVar
}
