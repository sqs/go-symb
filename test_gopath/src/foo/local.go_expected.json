[
  {
    "Expr": "foo",
    "Ident": "foo",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 8,
      "Line": 1,
      "Column": 9
    },
    "ExprType": "",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "Local": false,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "NonLocalVar",
    "Ident": "NonLocalVar",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 17,
      "Line": 3,
      "Column": 5
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 17,
      "Line": 3,
      "Column": 5
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "NonLocalVar",
      "Type": "int"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "NonLocalType",
    "Ident": "NonLocalType",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 39,
      "Line": 5,
      "Column": 6
    },
    "ExprType": "NonLocalType",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 39,
      "Line": 5,
      "Column": 6
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "NonLocalType",
      "Type": "NonLocalType"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "int",
    "Ident": "int",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 52,
      "Line": 5,
      "Column": 19
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": null,
      "Name": "int",
      "Type": "int"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "localRecv",
    "Ident": "localRecv",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 63,
      "Line": 7,
      "Column": 7
    },
    "ExprType": "*NonLocalType",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 63,
      "Line": 7,
      "Column": 7
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "localRecv",
      "Type": "*NonLocalType"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "NonLocalType",
    "Ident": "NonLocalType",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 74,
      "Line": 7,
      "Column": 18
    },
    "ExprType": "NonLocalType",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 39,
      "Line": 5,
      "Column": 6
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "NonLocalType",
      "Type": "NonLocalType"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "(*NonLocalType).NonLocalFunc",
    "Ident": "NonLocalFunc",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 88,
      "Line": 7,
      "Column": 32
    },
    "ExprType": "func(localParam int) (localResult int)",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 88,
      "Line": 7,
      "Column": 32
    },
    "ReferObj": {
      "Isa": "Func",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "NonLocalFunc",
      "Type": "func(localParam int) (localResult int)"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "localParam",
    "Ident": "localParam",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 101,
      "Line": 7,
      "Column": 45
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 101,
      "Line": 7,
      "Column": 45
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "localParam",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "int",
    "Ident": "int",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 112,
      "Line": 7,
      "Column": 56
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": null,
      "Name": "int",
      "Type": "int"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "localResult",
    "Ident": "localResult",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 118,
      "Line": 7,
      "Column": 62
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 118,
      "Line": 7,
      "Column": 62
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "localResult",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "int",
    "Ident": "int",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 130,
      "Line": 7,
      "Column": 74
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": null,
      "Name": "int",
      "Type": "int"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "localVar",
    "Ident": "localVar",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 142,
      "Line": 8,
      "Column": 6
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 142,
      "Line": 8,
      "Column": 6
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "localVar",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "int",
    "Ident": "int",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 151,
      "Line": 8,
      "Column": 15
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": null,
      "Name": "int",
      "Type": "int"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "println",
    "Ident": "println",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 160,
      "Line": 9,
      "Column": 2
    },
    "ExprType": "\u003ctype of println\u003e",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "Func",
      "Pkg": null,
      "Name": "println",
      "Type": "\u003ctype of println\u003e"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "NonLocalVar",
    "Ident": "NonLocalVar",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 168,
      "Line": 9,
      "Column": 10
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 17,
      "Line": 3,
      "Column": 5
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "NonLocalVar",
      "Type": "int"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "localRecv",
    "Ident": "localRecv",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 181,
      "Line": 9,
      "Column": 23
    },
    "ExprType": "NonLocalType",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 63,
      "Line": 7,
      "Column": 7
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "localRecv",
      "Type": "*NonLocalType"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "NonLocalType",
    "Ident": "NonLocalType",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 205,
      "Line": 9,
      "Column": 47
    },
    "ExprType": "NonLocalType",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 39,
      "Line": 5,
      "Column": 6
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "NonLocalType",
      "Type": "NonLocalType"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "localVar",
    "Ident": "localVar",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 222,
      "Line": 9,
      "Column": 64
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 142,
      "Line": 8,
      "Column": 6
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "localVar",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "localParam",
    "Ident": "localParam",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 232,
      "Line": 9,
      "Column": 74
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 101,
      "Line": 7,
      "Column": 45
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "localParam",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "localResult",
    "Ident": "localResult",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 244,
      "Line": 9,
      "Column": 86
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 118,
      "Line": 7,
      "Column": 62
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "localResult",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "localVar",
    "Ident": "localVar",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 265,
      "Line": 10,
      "Column": 9
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/local.go",
      "Offset": 142,
      "Line": 8,
      "Column": 6
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "localVar",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  }
]
